### ISSUE TITLE
functionObjects executed at start in OpenFOAM8

### ISSUE THREAD

**Issue by DavidSCN** [precice/openfoam-adapter#179]
In https://github.com/OpenFOAM/OpenFOAM-dev/commit/2ebed5ec7123ff8750931a65ab57a7bc7683aaa1, the default behavior of `functionObjects` has been changed and extended to that they are executed already before the simulation starts, i.e. the `execute` subroutine is called before we start a time step. 
In terms of our adapter, this means we write to and advance preCICE initially with the initial condition of our simulation. That might be in particular a problem (see also https://github.com/precice/openfoam-adapter/pull/130#issuecomment-831483248) if we have zero initial condition (initial field also with zero velocity) since our solid solver doesn't receive any loads during the first time window and (if everything goes well) calculates a zero displacement. I guess it's more or less undefined in terms of convergence (division by zero). However, preCICE raises a warning and emits convergence at this point (after one iteration)

```bash
---[precice]  relative convergence measure: relative two-norm diff = inf, limit = 0.005, normalization = 0, conv = true
---[precice]  relative convergence measure: relative two-norm diff = inf, limit = 0.005, normalization = 0, conv = true
---[precice]  All converged
---[precice] WARNING:  The coupling residual equals almost zero. There is maybe something wrong in your adapter. Maybe you always write the same data or you call advance without providing new data first or you do not use available read data. Or you just converge much further than actually necessary.
```

Afterwards, the simulation starts more or less as usual, we just lag one time step behind the correct time series. Still, the new default behavior is certainly not what we want and we actually have an option for writing initial data in the preCICE configuration in case this is desired by the user.

Luckily, the new default option is a public boolean: We can simply add the following line
```c++
    this->executeAtStart_ = false;
```
in the constructor of our `functionObject`

https://github.com/precice/openfoam-adapter/blob/aab511c0756f9dc50b542d2c9166502b5ee56a9d/preciceAdapterFunctionObject.C#L52-L61

Usually, the parameter is intended to be set at run time in the `controlDict` (it is one of the default parameters every functionObject reads), but we don't read these base class parameters at all, so disabling this option in the code by default should be just fine and doesn't require any changes in the configuration of the tutorial.

Any comment?

**Comment by MakisH** [comment]
Thank you very much for looking into this! Yes, if only such a small change is needed, we can definitely integrate it in the OpenFOAM8 branch.

**Comment by DavidSCN** [comment]
Yes, do you like to patch it directly in #130 or do you want me to take action?

**Comment by MakisH** [comment]
Please make a new branch `OpenFOAM8` from `OpenFOAM7` and open a PR there. Then we should also move the target ot #130 to `OpenFOAM8`. There are two individual changes (which we will then probably squash after merging).

**Comment by DavidSCN** [comment]
Closed via #180
